### PR TITLE
CODENVY-1597: Move integration properties from files to common config file

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Change the location of JIRA and JENKINS integration properties from separate files to `codenvy.env` file

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1597

### Changelog and Release Note Information
**Changelog**:
Added JIRA and Jenkins integration properties to `codenvy.env` file in an `INTEGRATION` section .

**Release Notes**:

**JIRA and Jenkins Integrations with BitBucket**
Codenvy can use [Factories]() to integrate with your issue management and CI systems. This allows us to generate developer workspaces pre-configured with the context of the issue or CI job (including project and branch). We have documented integrations with JIRA and Jenkins:

- JIRA: Jump directly from an issue in JIRA to a custom, isolated workspace designed to let a developer edit, built, debug and review the issue on the correct branch.
- Jenkins: A failed CI build email can be customized to include a link to a Codenvy Factory that will generate a workspace already tied to the repo, branch and commit ID that broke the build, simplifying diagnosis.

With this release we have added support for BitBucket Server (on-premises) in addition to the existing support for GitHub.

Configuration for this can be done in `codenvy.env` in a new section "Integrations." Each property is described in that file with a provided example.

```
# Connectors
# The association of third-party system events with the specific Factory that
# should be invoked by that event is an association made with a connector.
# There are different types of connectors that require different parameters.
# Replace <CONNECTORID> with a unique name for each connector that you create.
# The <CONNECTORID> can only contain standard characters.
# Example:
#   CODENVY_JENKINS_CONNECTOR_MYJENKINS_FACTORY_ID=r6p0l1sfnwm99k94
#   CODENVY_JENKINS_CONNECTOR_MYJENKINS_URL=http://userName:password@jenkins.codenvy-dev.com:8080
#   CODENVY_JENKINS_CONNECTOR_MYJENKINS_JOB_NAME=test

# Webhooks
# An integrations may require a webhook into the third-party system
# to register and capture events to automate actions.
# Replace <WEBHOOKID> with a unique name for each webhook that you create.
# The <WEBHOOKID> can only contain standard characters. You can include as many
# Factories with each webhook as you desire.
# Example:
#   CODENVY_GITHUB_WEBHOOK_MYJENKINS_URL=github
#   CODENVY_GITHUB_WEBHOOK_MYJENKINS_FACTORY1_ID=r6p0l1sfnwm99k94
#   CODENVY_GITHUB_WEBHOOK_MYJENKINS_FACTORY2_ID=g8n2t3lsidu54z88
```
If you'd like to speak to us about an integrations between Codenvy and another CI system, [please contact us](https://codenvy.com/contact/questions/).

### Docs Pull Request
https://github.com/codenvy/codenvy-docs/pull/60

### CODE REVIEW

File: dockerfiles/init/modules/codenvy/templates/codenvy.env.erb [hunk 1]
```diff
@@ -69,3 +69,24 @@ JAVA_OPTS=-Xms512m -Xmx<%= @codenvy_server_xmx %>m -server -XX:+HeapDumpOnOutOfM
...
+<%= k + '=' + v %>
+<% end -%>
+<% end -%>
+
+# integration Bitbucket Server webhooks
+<% ENV.each do |k,v| -%>
+<% if k.include? "CODENVY_BITBUCKET_SERVER_" then  -%>
+<%= k + '=' + v %>
+<% end -%>
+<% end -%>
```
Comment (riuvshin) [inline dockerfiles/init/modules/codenvy/templates/codenvy.env.erb]: please add new line

File: plugins/plugin-bitbucket/codenvy-plugin-bitbucketserver-webhooks/src/main/java/com/codenvy/plugin/webhooks/bitbucketserver/BitbucketServerWebhookService.java [hunk 2]
```diff
@@ -157,34 +164,24 @@ private String computeCloneUrl(String owner, String projectKey, String repositor
...
+
+        Set<String> webhooks = properties.entrySet()
+                                         .stream()
+                                         .filter(entry -> repositoryUrl.equals(entry.getValue()))
+                                         .map(entry -> entry.getKey()
+                                                            .substring(0, entry.getKey().lastIndexOf(WEBHOOK_REPOSITORY_URL_SUFFIX)))
+                                         .collect(toSet());
+
+        if (webhooks.isEmpty()) {
+            LOG.error("No BitBucket Server webhooks were registered for repository {}", repositoryUrl);
```
Comment (tolusha) [inline plugins/plugin-bitbucket/codenvy-plugin-bitbucketserver-webhooks/src/main/java/com/codenvy/plugin/webhooks/bitbucketserver/BitbucketServerWebhookService.java]: If it is an error, should we throw an exception ?

File: dockerfiles/init/modules/codenvy/templates/codenvy.env.erb (hunk 1 above)
Comment (vinokurig) [inline dockerfiles/init/modules/codenvy/templates/codenvy.env.erb]: done

File: plugins/plugin-bitbucket/codenvy-plugin-bitbucketserver-webhooks/src/main/java/com/codenvy/plugin/webhooks/bitbucketserver/BitbucketServerWebhookService.java (hunk 2 above)
Comment (vinokurig) [inline plugins/plugin-bitbucket/codenvy-plugin-bitbucketserver-webhooks/src/main/java/com/codenvy/plugin/webhooks/bitbucketserver/BitbucketServerWebhookService.java]: It is not an error, changed th log level to warnong

**Comment by bmicklea** [comment]
@vinokurig I removed the "<" and ">" from your variable names because I want to make sure that if a user doesn't read the explanation and just uncomments the variable, it will work.

**Comment by benoitf** [comment]
typo has not been fixed in Release Notes section
Release Notes:
Jira and `F`enkins

**Comment by bmicklea** [comment]
Updating release notes now.

**Comment by bmicklea** [comment]
@vinokurig please review the Release Notes section in the PR description to make sure my examples are correct.

**Comment by vinokurig** [comment]
@bmicklea removed the `BitBucket` from:
>With this release we have added support for BitBucket Server (on-premises) in addition to the existing support for GitHub and Bitbucket.

We do not support integration for Bitbucket

**Pull request merged**